### ISSUE TITLE
Bug: class decorator parsed as unused

### ISSUE THREAD

**Issue by woody-li** [eslint/eslint#19045]
### Environment

Node version: 22
npm version: 10.9.0
Local ESLint version: 9.13.0
Global ESLint version:
Operating System: Win 11

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
import globals from 'globals';
import pluginJs from '@eslint/js';
import pluginReact from 'eslint-plugin-react';
import babelParser from '@babel/eslint-parser';
export default [
    pluginJs.configs.recommended,
    pluginReact.configs.flat.recommended,
    {
        languageOptions: {
            globals: globals.browser,
            parser: babelParser,
            parserOptions: {
                requireConfigFile: false,
                babelOptions: {
                    plugins: [['@babel/plugin-proposal-decorators', { version: '2023-11' }]],
                },
                ecmaFeatures: {
                    jsx: true,
                },
            },
        },
    },
    
];
```
</details>

```js
function define(name){
    function init() {
        _define(name, this); // do sth.
    }
    return (value, { addInitializer }) => addInitializer(init);
}

@define('button')
class Button{
}
```

### What did you expect to happen?

With decorator, the class is used as other logic.
So it shouldn't be unused.

### What actually happened?

It parsed as unused.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @woody-li, thanks for the issue!

By the issue title, I assume you're getting an error from the `no-unused-vars` rule.

As described in our [policies around experimental features](https://github.com/eslint/eslint#what-about-experimental-features), ESLint core rules only support the latest ECMAScript standard. The [Decorators proposal](https://github.com/tc39/proposal-decorators) is still in Stage 3. If and when the proposal reaches Stage 4, we will update core rules to support this syntax.

In the meantime, you can use [eslint-disable](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) comments to suppress these errors, or ask for a replacement rule in [`@babel/eslint-plugin`](https://github.com/babel/babel/tree/main/eslint/babel-eslint-plugin).
